### ISSUE TITLE
get_variable with a file object as default value: Argument of type File is not held by an ObjectHolder

### ISSUE THREAD

**Issue by phillipjohnston** [mesonbuild/meson#8977]
**Describe the bug**
After updating Meson, I see this error in a previously working build:

```
build/analysis/vale/meson.build:24:0: ERROR: Argument build/analysis/vale/vale-styleguide/config/documentation.vale.ini of type File is not held by an ObjectHolder.

    This is a Meson bug and should be reported!
```

The file is being specified in this manner:

```
# Supply a style file, which will use this file instead of the default .vale.ini
vale_config_file = get_variable('vale_config_file',
	files('vale-styleguide/config/documentation.vale.ini'))
```

The default variable option is being used - I'm not overriding it.

The same is happening in a Doxygen module I use:

```
doxyfile_input = get_variable('doxyfile_input', files('Doxyfile.in'))
```

I tried moving the file object into another variable:

```
vale_default_config_file = files('vale-styleguide/config/documentation.vale.ini')
vale_config_file = get_variable('vale_config_file', vale_default_config_file)
```

With teh same result - the error is reported on the `get_variable` line.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? **native**
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.) **MacOS 10.15.7**
* what Python version are you using e.g. 3.8.0 **Python 3.9.6**
* what `meson --version` **0.59.0.rc1**

**Comment by phillipjohnston** [comment]
Went back to 0.58.1 - the error is not present in that build.

**Comment by mensinda** [comment]
Should be fixed with #8978
